tests/mixed_int8/README.md

```diff
@@ -1,37 +1,122 @@
+- T4 + (e.g. Google Colab GPUs) 
 
 ## Virutal envs
 
 ```conda create --name int8-testing python==3.8```
-```git clone https://github.com/younesbelkada/transformers.git && git checkout integration-8bit```
-```pip install -e ".[dev]"```
-```pip install -i https://test.pypi.org/simple/ bitsandbytes```
-```pip install git+https://github.com/huggingface/accelerate.git@e0212893ea6098cc0a7a3c7a6eb286a9104214c1```
-
+```pip install bitsandbytes>=0.31.5```
+```pip install accelerate>=0.12.0```
+```pip install git+https://github.com/huggingface/transformers.git```
+if `transformers>=4.22.0` is not released yet, otherwise
+```pip install transformers```
 
 ## Trobleshooting
 
+A list of common errors:
+
+### Torch does not correctly do the operations on GPU
+
+First check that:
+
+```py
+import torch
+
+vec = torch.randn(1, 2, 3).to(0)
+```
+
+Works without any error. If not install torch using `conda` like:
+
 ```conda create --name int8-testing python==3.8```
-```pip install -i https://test.pypi.org/simple/ bitsandbytes```
+```pip install bitsandbytes```
 ```conda install pytorch torchvision torchaudio -c pytorch```
-```git clone https://github.com/younesbelkada/transformers.git && git checkout integration-8bit```
-```pip install -e ".[dev]"```
-```pip install git+https://github.com/huggingface/accelerate.git@b52b793ea8bac108ba61192eead3cf11ca02433d```
+```pip install git+https://github.com/huggingface/transformers.git```
+```pip install accelerate```
+
+And the snippet above should work
+
+### ` `bitsandbytes` operations are not supported under CPU!`
+
+This happens when some Linear weights are set to the CPU when using `accelerate`. Please check carefully `model.hf_device_map` and make sure that there is no `Linear` module that is assigned to CPU. It is fine to have the last module (usually the Lm_head) set on CPU.
+
+### `To use the type as a Parameter, please correct the detach() semantics defined by __torch_dispatch__() implementation.`
+
+Use the latest version of `accelerate` with a command such as: `pip install --force accelerate` and the problem should be solved.
 
-### Check driver settings:
+### `Parameter has no attribue .CB` 
 
+Same comment as above
+
+### `RuntimeError: CUDA error: an illegal memory access was encountered ... consider passing CUDA_LAUNCH_BLOCKING=1`
+
+Run your script by pre-pending `CUDA_LAUNCH_BLOCKING=1` and you should observe an error as below:
+
+### `CUDA illegal memory error: an illegal memory access at line...`:
+
+Check the CUDA verisons with:
 ```
 nvcc --version
 ```
-
+And confirm it is the same version than the one detected by `bitsandbytes`. If not, run:
 ```
 ls -l $CONDA_PREFIX/lib/libcudart.so
 ```
+or 
+```
+ls -l $LD_LIBRARY_PATH
+```
+And check if `libcudart.so` has a correct symlink that is set. Sometimes `nvcc` detects the correct CUDA version but `bitsandbytes` doesn't. You have to make sure that the symlink that is set for the file `libcudart.so` is redirected to the correct CUDA file. 
```

Comment:
you can use `and`, or just `Check if ...`

Reply:
Perfect! Fixed it in c8d628195bd9fca4ae25031205d220aa2248ab70